app/services/hyrax/access_control_list.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module Hyrax
+  ##
+  # ACLs for `Hyrax::Resource` models
+  #
+  # Allows managing `Hyrax::Permission` entries referring to a specific
+  # `Hyrax::Resource` using a simple add/delete model.
+  class AccessControlList
+    ##
+    # @!attribute [rw] resource
+    #   @return [Valkyrie::Resource]
+    # @!attribute [r] persister
+    #   @return [#save]
+    # @!attribute [r] query_service
+    #   @return [#find_inverse_references_by]
+    attr_reader :persister, :query_service
+    attr_accessor :resource
+
+    ##
+    # @param resource [Valkyrie::Resource]
+    # @param persister [#save]
+    # @param query_service [#find_inverse_references_by]
+    def initialize(resource:, persister: Hyrax.persister, query_service: Hyrax.query_service)
+      self.resource  = resource
+      @persister     = persister
+      @query_service = query_service
+    end
+
+    ##
+    # @param permission [Hyrax::Permission]
+    #
+    # @return [Boolean]
+    def <<(permission)
+      permission.access_to = resource.id
+
+      additions << permission
+
+      true
```

Comment:
Why return boolean value instead of `additions` array?

Reply:
`additions` and `deletions` are internal implementation details that i don't expect to survive sustained development. it seems likely this code will use a `ChangeSet` or similar eventually.

For this reason, exposing them via the API seemed fragile.